src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -201,54 +213,44 @@ internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSynt
-                if (allSymbolReferences.Count >= MaxResults)
+                // Ignore netmodules.  First, they're incredibly esoteric and barely used.
+                // Second, the SymbolFinder api doesn't even support searching them. 
+                var assembly = compilation.GetAssemblyOrModuleSymbol(reference) as IAssemblySymbol;
+                if (assembly != null)
                 {
-                    break;
+                    findTasks.Add(finder.FindInMetadataAsync(project.Solution, assembly, reference, exact));
                 }
             }
-        }
 
-        private async Task FindResultsInMetadataReferences(
-            Project otherProject,
-            List<SymbolReference> allSymbolReferences,
-            SymbolReferenceFinder finder,
-            HashSet<PortableExecutableReference> seenReferences,
-            bool exact,
-            CancellationToken cancellationToken)
-        {
-            // See if this project has a metadata reference we haven't already looked at.
-            var newMetadataReferences = otherProject.MetadataReferences.OfType<PortableExecutableReference>();
-
-            Compilation compilation = null;
-            foreach (var reference in newMetadataReferences)
+            while (findTasks.Count > 0)
             {
-                // Make sure we don't check the same metadata reference multiple times from 
-                // different projects.
-                if (seenReferences.Add(reference))
-                {
-                    // Defer making the compilation until necessary.
-                    compilation = compilation ?? await otherProject.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+                // Keep on looping through the 'find' tasks, processing each when they finish.
+                cancellationToken.ThrowIfCancellationRequested();
+                var doneTask = await Task.WhenAny(findTasks).ConfigureAwait(false);
 
-                    // Ignore netmodules.  First, they're incredibly esoteric and barely used.
-                    // Second, the SymbolFinder api doesn't even support searching them. 
-                    var assembly = compilation.GetAssemblyOrModuleSymbol(reference) as IAssemblySymbol;
-                    if (assembly != null)
-                    {
-                        AddRange(allSymbolReferences, await finder.FindInMetadataAsync(otherProject.Solution, assembly, reference, exact).ConfigureAwait(false));
-                    }
-                }
+                // One of the tasks finished.  Remove it from the list we're waiting on.
+                findTasks.Remove(doneTask);
+
+                // Add its results to the final result set we're keeping.
+                AddRange(allSymbolReferences, await doneTask.ConfigureAwait(false));
 
+                // If we've got enough, no need to keep searching. 
+                // Note: Should we cancel the existing work?  IMO, no.  These tasks will
```

Comment:
Consider rephrasing to just stating the designed behavior and the justification.